Dodge.js

```diff
@@ -0,0 +1,179 @@
+import { StatusBar } from 'expo-status-bar';
+import { StyleSheet, Text, View, Dimensions, Button, Image } from 'react-native';
+import { Knight, Circle, Brick, Grenade, MoveKnight, Shrapnel, counter} from "./Entities.js";
+import { GameEngine } from "react-native-game-engine";
+import Matter from "matter-js";
+import AxisPad from 'react-native-axis-pad';
+////this looks interesting for the attacker placing/throwing projectiles:
+//https://reactnative.dev/docs/panresponder 
+//https://medium.com/yapsody-engineering/hooks-and-function-components-in-react-native-d4e667c90cda
+var ents = {}
+
+const generateUniqueId = require('generate-unique-id');
+
+const MAXwidth = Dimensions.get('window').width;
+const MAXheight = Dimensions.get('window').height;
+
+
+//adds an entity given its type, if it's a Grenade set a timer for explosion
+function AddEntity(entity){
+let id = generateUniqueId()
+ents[id] = new entity(id)
+
+if(entity == Grenade){
+  setTimeout(() => { Explode(id) }, 2000)
+}
+}
+
+//deletes a single entity given that entities unique id
+function DeleteEntity(id){
+delete(ents[id])
```

Comment:
Indentation, in like a million places :P 
Remember your code editor should be able to "format document" or something like that.